base/server/src/com/netscape/cmscore/apps/CMSEngine.java

```diff
@@ -1076,6 +1096,17 @@ public void start() throws Exception {
         mStartupTime = System.currentTimeMillis();
 
         logger.info(name + " engine started");
+
+        for (StartupNotifier sink : startupNotifiers) {
+            try {
+                sink.notifyReady();
```

Comment:
To avoid duplicate notifications maybe we could use Tomcat's `LifecycleListener` which
is added into server.xml instead of CS.cfg:
https://tomcat.apache.org/tomcat-9.0-doc/api/org/apache/catalina/LifecycleListener.html
https://github.com/dogtagpki/pki/blob/master/base/server/tomcat-8.5/conf/server.xml#L86
https://github.com/dogtagpki/pki/blob/master/base/server/tomcat/src/com/netscape/cms/tomcat/PKIListener.java
https://github.com/dogtagpki/tomcatjss/blob/master/src/org/dogtagpki/tomcat/JSSListener.java

Reply:
Will look into `LifecycleListener` today, thanks.

The current implementation - the notify behaviour is in CMSEngine but the notifiers are configured explicitly in the subsystem. So you just configure it in `ca/CS.cfg` and that's that.